databases/week1/week1.sql

```diff
@@ -0,0 +1,60 @@
+--Find out how many tasks are in the task table
+SELECT COUNT(id) as total_task FROM task;
+
+--Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(id)
+FROM task
+WHERE due_date IS NULL;
+
+--Find all the tasks that are marked as done
+SELECT * 
+FROM task JOIN status
+ON task.status_id = status_id
+WHERE status.name ="Done";
+
+--Find all the tasks that are not marked as done
+SELECT *
+FROM task JOIN status
+ON task.status_id = status_id
+WHERE status.name != "Done";
+
+--Get all the tasks, sorted with the most recently created first
+SELECT *
+FROM task
+ORDER BY created DESC;
+
+--Get the single most recently created task
+SELECT *
+FROM task
+ORDER BY created DESC
+LIMIT 1;
+
+--Get the title and due date of all tasks where the title or description contains database
+
+SELECT task.title, task.due_date, task.description
+FROM task
+WHERE title
+LIKE "%database%" 
+OR description LIKE "%database%" ;
```

Comment:
Query formatting:
SELECT 
task.title,
task.due_date,
task.description
FROM task
WHERE title LIKE "%database%" or description LIKE "%databases%"

Pointer: keeping the where conditions in one liner when possible, improves readability.